Samples/ExtendedControlAndMetadata/EyeGazeAndBackgroundSegmentation/CameraKsPropertyHelper/PropertyInquiry.cpp

```diff
@@ -0,0 +1,183 @@
+                break;
+            }
+            
+
+            default:
+                throw hresult_invalid_argument(L"unexpected extendedControlKind passed: " + to_hstring((int32_t)extendedControlKind));
+        }
+
+        return payload;
+    }
+
+    /// <summary>
+    /// Send a SET command of a supported extended control onto the specified VideoDeviceController with the specified flags value
+    /// </summary>
+    /// <param name="controller"></param>
+    /// <param name="extendedControlKind"></param>
+    /// <param name="flags"></param>
+    void PropertyInquiry::SetExtendedControlFlags(
+        Windows::Media::Devices::VideoDeviceController const& controller,
+        winrt::CameraKsPropertyHelper::ExtendedControlKind const& extendedControlKind,
+        uint64_t flags)
+    {
+        int controlId = static_cast<int>(extendedControlKind);
+        
+        KSPROPERTY prop(
+            { 0x1CB79112, 0xC0D2, 0x4213, {0x9C, 0xA6, 0xCD, 0x4F, 0xDB, 0x92, 0x79, 0x72} } /* KSPROPERTYSETID_ExtendedCameraControl */,
+            controlId /* KSPROPERTY_CAMERACONTROL_EXTENDED_* */,
+            0x00000002 /* KSPROPERTY_TYPE_SET */);
+        uint8_t* pProp = reinterpret_cast<uint8_t*>(&prop);
+        array_view<uint8_t const> serializedProp = array_view<uint8_t const>(pProp, sizeof(KSPROPERTY));
+
+        KSCAMERA_EXTENDEDPROP_HEADER value =
+        {
+            1, // Version
+            1, // PinId = KSCAMERA_EXTENDEDPROP_FILTERSCOPE
+            sizeof(KSCAMERA_EXTENDEDPROP_HEADER) + sizeof(KSCAMERA_EXTENDEDPROP_VALUE), // Size
+            0, // Result
+            flags, // Flags
+            0 // Capability                
+        };
+        uint8_t* pValue = reinterpret_cast<uint8_t*>(&value);
+        array_view<uint8_t const> serializedValue = array_view<uint8_t const>(pValue, value.Size);
+
+        auto setResult = controller.SetDevicePropertyByExtendedId(serializedProp, serializedValue);
+
+        if (setResult != Windows::Media::Devices::VideoDeviceControllerSetDevicePropertyStatus::Success)
+        {
+            throw hresult_invalid_argument(L"Could not set device property flags " + winrt::to_hstring(controlId) + L" with status: " + winrt::to_hstring((int)setResult));
+        }
+    }
+
+    /// <summary>
+    /// Extract the frame metadata of a supported kind from the specified MediaFrameReference
+    /// </summary>
+    /// <param name="frame"></param>
+    /// <param name="metadataKind"></param>
+    /// <returns></returns>
+    winrt::CameraKsPropertyHelper::IMetadataPayload PropertyInquiry::ExtractFrameMetadata(winrt::Windows::Media::Capture::Frames::MediaFrameReference const& frame, winrt::CameraKsPropertyHelper::FrameMetadataKind const& metadataKind)
+    {
+        static guid captureMetadataGuid = guid(0x2EBE23A8, 0xFAF5, 0x444A, { 0xA6, 0xA2, 0xEB, 0x81, 0x08, 0x80, 0xAB, 0x5D }); // MFSampleExtension_CaptureMetadata
```

Comment:
>MFSampleExtension_CaptureMetadata [](http://example.com/codeflow?start=131&length=33)

can we get the guids from standard headers?

Reply:
we could if we were building with the minimum SDK version that has those defined in it.